dpe/src/context.rs

```diff
@@ -251,6 +255,95 @@ impl<'a> Iterator for ChildToRootIter<'a> {
     }
 }
 
+pub(crate) struct RootToChildIter<'a> {
+    contexts: &'a [Context],
+    path: [u8; MAX_HANDLES],
+    path_size: usize,
+    done: bool,
+    curr: usize,
+}
+
+impl<'a> RootToChildIter<'a> {
+    /// Create a new iterator that will start at the leaf and go to the root node.
+    pub fn new(leaf_idx: usize, contexts: &'a [Context]) -> Result<Self, DpeErrorCode> {
+        let mut iter = ChildToRootIter::new(leaf_idx, contexts);
+        let mut path = [0u8; MAX_HANDLES];
+        let mut path_size = 0;
+
+        // Use while loop to avoid consuming iter so curr_idx can be read
+        // inside the loop
+        let mut curr = iter.curr_idx();
+        while let Some(_) = iter.next() {
+            if path_size > contexts.len() {
+                return Err(DpeErrorCode::InternalError);
+            }
+
+            // Guaranteed to fit into u8 since Contexts struct is of size
+            // MAX_HANDLES.
+            path[path_size] = curr as u8;
+            path_size += 1;
+            curr = iter.curr_idx();
+        }
+
+        Ok(Self {
+            contexts,
+            path,
+            path_size,
+            done: false,
+            curr: path_size - 1,
```

Comment:
We should probably check `path_size` is >=1 before this. What would happen if contexts is empty?

Reply:
Done